packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts

```diff
@@ -0,0 +1,219 @@
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { AdapterResponse, makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import { ethers } from 'ethers'
+import { hex } from '@scure/base'
+import { p2tr, p2tr_ns, taprootTweakPubkey, TAPROOT_UNSPENDABLE_KEY } from '@scure/btc-signer'
+import { BaseEndpointTypes } from '../endpoint/proof-of-reserves'
+import abi from '../config/dlc-manager-abi.json'
+import {
+  BitcoinNetwork,
+  BitcoinTransaction,
+  BitcoinTransactionVectorOutput,
+  getBitcoinNetwork,
+  RawVault,
+} from './utils'
+
+const logger = makeLogger('dlcBTC PoR')
+
+export type TransportTypes = BaseEndpointTypes
+
+export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<TransportTypes>
+  requester!: Requester
+  provider!: ethers.providers.JsonRpcProvider
+  dlcManagerContract!: ethers.Contract
+  settings!: TransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<TransportTypes>,
+    adapterSettings: TransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    const { RPC_URL, CHAIN_ID, DLC_CONTRACT } = adapterSettings
+    this.settings = adapterSettings
+    this.provider = new ethers.providers.JsonRpcProvider(RPC_URL, CHAIN_ID)
+    this.dlcManagerContract = new ethers.Contract(DLC_CONTRACT, abi, this.provider)
+    this.requester = dependencies.requester
+  }
+  async backgroundHandler(context: EndpointContext<TransportTypes>) {
+    await this.handleRequest()
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest() {
+    let response: AdapterResponse<TransportTypes['Response']>
+    try {
+      response = await this._handleRequest()
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      logger.error(e, errorMessage)
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: [], response }])
+  }
+
+  async _handleRequest(): Promise<AdapterResponse<TransportTypes['Response']>> {
+    const providerDataRequestedUnixMs = Date.now()
+
+    // Get all vault data. Filter placeholder values.
+    const vaultData: RawVault[] = (await this.dlcManagerContract.getFundedDLCs(0, 10_000)).filter(
```

Comment:
this means we are capped at 10k vaults, and 1 address can have multiple vaults, right?
I guess the question is dependent on Joshua's answer to [this](https://smartcontract-it.atlassian.net/browse/DF-19830?focusedCommentId=229460), whether we should preemptively add pagination here or maybe even ask them to maintain a numFundedDLCs.

Reply:
Yes, we are capped at 10k vaults, tho now i think maybe it makes sense to makes this as a config as well. I think they do track the number of [vaults](https://github.com/DLC-link/dlc-solidity/blob/dev/contracts/DLCManager.sol#L267) but it's a private property.

Reply:
Depending on how the RPC handles the load we might need to ask them to expose it and send paginated requests in batch to get the full list of vaults

Reply:
Probably best to ask asap then since it will require redeploying contracts